packages/devtools-reps/src/object-inspector/utils.js

```diff
@@ -227,70 +348,105 @@ function makeNodesForProperties(objProps, parent, { bucketSize = 100 } = {}) {
   const parentValue = getValue(parent);
 
   let allProperties = Object.assign({}, ownProperties, safeGetterValues);
-  const properties = sortProperties(Object.keys(allProperties)).filter(name =>
+
+  // Ignore properties that are neither non-concrete nor getters/setters.
+  const propertiesNames = sortProperties(Object.keys(allProperties)).filter(name =>
     allProperties[name].hasOwnProperty("value")
     || allProperties[name].hasOwnProperty("getterValue")
     || allProperties[name].hasOwnProperty("get")
     || allProperties[name].hasOwnProperty("set")
   );
 
-  const numProperties = properties.length;
+  const numProperties = propertiesNames.length;
 
   let nodes = [];
-  if (nodeIsArray(prototype) && numProperties > bucketSize) {
+  if (nodeSupportsBucketing(parent) && numProperties > bucketSize) {
     nodes = makeNumericalBuckets(
-      properties,
+      propertiesNames,
       bucketSize,
-      parentPath,
+      parent,
       allProperties
     );
-  } else if (parentValue.class == "Window") {
-    nodes = makeDefaultPropsBucket(properties, parentPath, allProperties);
+  } else if (parentValue && parentValue.class == "Window") {
+    nodes = makeDefaultPropsBucket(propertiesNames, parent, allProperties);
   } else {
-    nodes = makeNodesForOwnProps(properties, parentPath, allProperties);
+    nodes = makeNodesForOwnProps(propertiesNames, parent, allProperties);
   }
 
-  for (let index in ownSymbols) {
-    nodes.push(
-      createNode(
-        ownSymbols[index].name,
-        `${parentPath}/##symbol-${index}`,
-        ownSymbols[index].descriptor
-      )
-    );
+  if (Array.isArray(ownSymbols)) {
+    ownSymbols.forEach((ownSymbol, index) => {
+      nodes.push(
+        createNode(
+          parent,
+          ownSymbol.name,
+          `${parentPath}/${SAFE_PATH_PREFIX}symbol-${index}`,
+          ownSymbol.descriptor
+        )
+      );
+    }, this);
   }
 
-  if (isPromise(parent)) {
-    nodes.push(...getPromiseProperties(parent));
+  if (nodeIsPromise(parent)) {
+    nodes.push(...makeNodesForPromiseProperties(parent));
   }
 
   // Add the prototype if it exists and is not null
   if (prototype && prototype.type !== "null") {
     nodes.push(
-      createNode("__proto__", `${parentPath}/__proto__`, { value: prototype })
+      createNode(parent, "__proto__", `${parentPath}/__proto__`, { value: prototype })
     );
   }
 
   return nodes;
 }
 
-function createNode(name, path, contents) {
+function createNode(
+  parent: ObjectInspectorItem,
+  name: string,
+  path: string,
+  contents: any,
+  type: ?Symbol = NODE_TYPES.GRIP
```

Comment:
This represents the regular nodes ( = a grip), as opposed to all the other custom ones. I wanted to make it optional so we don't have to pass it when creating those regular nodes.